syntax = "proto2";
package rare.protos;

message RnnCell {
  oneof rnn_cell_oneof {
    LstmCell lstm_cell = 1;
    GruCell gru_cell = 2;
  }
}

message LstmCell {
  optional uint32 num_units = 1 [default=128];
  optional boolean use_peepholes = 2 [default=false];
  optional float32 forget_bias = 3 [default=1.0];
}

message GruCell {
  optional uint32 num_units = 1 [default=128];
}
